src/Tests/Microsoft.Diagnostics.Monitoring.ExecuteActionApp/Microsoft.Diagnostics.Monitoring.ExecuteActionApp.csproj

```diff
@@ -1,10 +1,14 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net6.0</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="xunit.assert" Version="$(XunitAssertVersion)" />
   </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Diagnostics.Monitoring.Tool.UnitTests\Microsoft.Diagnostics.Monitoring.Tool.UnitTests.csproj" />
```

Comment:
This is causing the failures due to the mismatch xunit.assert dependencies. The Microsoft.Diagnostics.Monitoring.Tool.UnitTests assembly references 2.4.2.0 (even though we ask it to build against 2.4.1.0); it's probably upgraded because it is known to be a unit test project. However, the ExecuteActionApp is not a test project, so it'll use whatever version we tell it to use, which is 2.4.1.0.

I think you should remove this ProjectReference and either move whatever common information that is needed to TestCommon or just link it in (the latter is probably easier).